src/UI/Components/Diagram/Context/EventWrapper.tsx

```diff
@@ -0,0 +1,189 @@
+ *
+ * @props {React.PropsWithChildren} props - The properties that define the behavior and display of the component.
+ * @prop {React.ReactNode} children - The children components to be wrapped.
+ *
+ * @returns {React.FC<React.PropsWithChildren>} The EventWrapper component.
+ */
+export const EventWrapper: React.FC<React.PropsWithChildren> = ({
+  children,
+}) => {
+  const {
+    setStencilState,
+    setServiceOrderItems,
+    setCellToEdit,
+    setDictToDisplay,
+    looseEmbedded,
+    setLooseEmbedded,
+  } = useContext(CanvasContext);
+
+  /**
+   * Handles the event triggered when there are loose embedded entities on the canvas.
+   * The loose embedded entities are the entities that are not connected to the main entity. If there are any, they will be highlighted and deploy button will be disabled.
+   *
+   * @param {CustomEvent} event - The event object.
+   *
+   * @returns {void}
+   */
+  const handleLooseEmbeddedEvent = (event): void => {
+    const customEvent = event as CustomEvent;
+    const eventData: { kind: EmbeddedEventEnum; id: string } = JSON.parse(
+      customEvent.detail,
+    );
+    const newSet = new Set(looseEmbedded);
+
+    if (eventData.kind === "remove") {
+      newSet.delete(eventData.id);
+    } else {
+      newSet.add(eventData.id);
+    }
+    setLooseEmbedded(newSet);
+  };
+
+  /**
+   * Handles the event triggered when the user wants to see the dictionary properties of an entity.
+   *
+   * @param {CustomEvent} event - The event object.
+   *
+   * @returns {void}
+   */
+  const handleDictEvent = (event): void => {
+    const customEvent = event as CustomEvent;
+
+    setDictToDisplay(JSON.parse(customEvent.detail));
+  };
+
+  /**
+   * Handles the event triggered when the user wants to edit an entity.
+   *
+   * @param {CustomEvent} event - The event object.
+   *
+   * @returns {void}
+   */
+  const handleEditEvent = (event): void => {
+    const customEvent = event as CustomEvent;
+
+    setCellToEdit(customEvent.detail);
+  };
+
+  /**
+   * Handles the event triggered when the user made update to the instance cell
+   * we need to assert first if the instance triggered the event is in the serviceOrderItems map,
+   * as the related instances aren't added to the serviceOrderItems map as we don't want to accidentally delete or edit them,
+   * yet they can exist on the canvas and be removed from it.
+   *
+   * @param {CustomEvent} event - The event object.
+   *
+   * @returns {void}
+   */
+  const handleUpdateServiceOrderItems = (event): void => {
+    const customEvent = event as CustomEvent;
+    const { cell, action } = customEvent.detail as {
+      cell: ServiceEntityBlock;
+      action: ActionEnum;
+    };
+
+    setServiceOrderItems((prev) => {
+      // related instances aren't added to the serviceOrderItems map, this condition is here to prevent situation where we try to remove the related instance from the canvas and it ends up here with status to delete it from the inventory
```

Comment:
here too, be more explicit.  what are the related instances?

Reply:
I think it's quite explicit, especially that relation in app is I think exclusively related(no pun intended) to inter-service relation

Reply:
Well, that's the thing, sometimes related is used for embedded or inter-service-relations. We have an explicit term, so we should use it to avoid any possible confusion. If we had no term for them, sure, but we have the right terms. There's no reason to reenvent them :)

Reply:
Is it used though? I never came across that use but, maybe I missed something, I'll change it :)